src/build/release.rs

```diff
@@ -18,45 +22,87 @@ pub fn release(p: &crate::project::Project, version: &String) -> Result<(), Erro
         fs::copy(file, format!("releases/{}/@{}/{}", version, modname, file))?;
     }
 
-    // Generate key
     if !Path::new("releases/keys").exists() {
         fs::create_dir("releases/keys")?;
     }
+
+    let mut key;
     let keyname = p.get_keyname();
-    if !Path::new(&format!("releases/keys/{}.bikey", keyname)).exists() {
-        println!("    {} {}.bikey", "KeyGen".green().bold(), keyname);
-        armake2::sign::cmd_keygen(PathBuf::from(&keyname))?;
-        fs::rename(format!("{}.bikey", keyname), format!("releases/keys/{}.bikey", keyname))?;
-        fs::rename(format!("{}.biprivatekey", keyname), format!("releases/keys/{}.biprivatekey", keyname))?;
+    // Generate and store key if required
+    if p.reuse_private_key.unwrap_or(false) {
+
+        // Make a new keypair if there isn't one already
+        if !Path::new(&format!("releases/keys/{}.bikey", keyname)).exists() {
+            println!("    {} {}.bikey", "KeyGen".green().bold(), keyname);
+
+            // Generate and write the keypair to disk in the current directory
+            cmd_keygen(PathBuf::from(&keyname))?;
+            fs::rename(
+                format!("{}.bikey", keyname),
+                format!("releases/keys/{}.bikey", keyname),
+            )?;
+            fs::rename(
+                format!("{}.biprivatekey", keyname),
+                format!("releases/keys/{}.biprivatekey", keyname),
+            )?;
+        }
+
+        // Read the private key from disk
+        key = BIPrivateKey::read(
```

Comment:
I am not a fan of this formatting at all, is this what rustfmt did?

Reply:
Yeah, I wouldn't have done this. It's before I increased the line length in the other PR so it'll probably disappear if I run it again.